frontend/apps/reader/modules/readerlink.lua

```diff
@@ -23,6 +23,7 @@ local T = ffiutil.template
 
 local ReaderLink = InputContainer:extend{
     location_stack = nil, -- table, per-instance
+    _external_buttons = {}
```

Comment:
Oh you know what... I did have this variable initialized in init. But I found that the buttons added from elsewhere got overwritten by the init. Should I use two tables? One for default buttons (initialized in ReaderLink), and a second for other buttons?

Reply:
That sounds like either a code or a design issue ;).

Reply:
The name of the variable implies that this should only contain extra buttons already, which means the code actually building the widgets shouldn't solely rely on it?

Reply:
> But I found that the buttons added from elsewhere got overwritten by the init

That might be because you are/were using ReaderLink:addToExternalLinkDialog instead of self.ui.link:addToExternalLinkDialog , so adding them to the class object instead of the instance (ie. in wallabag, 2 comments below this one).

Reply:
> The name of the variable implies that this should only contain extra buttons already

Depends on how you read that :) I am reading it as "external_link_buttons" - not as "external_external_link_buttons" :)

Reply:
Nevermind. I can just leave it nil and init the table when the first button is added.